gradle-suppressible-error-prone/src/main/java/com/palantir/gradle/suppressibleerrorprone/modes/common/ModifyCheckApiOption.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.suppressibleerrorprone.modes.common;
+
+import com.palantir.gradle.suppressibleerrorprone.modes.common.ModifyCheckApiOption.DoNotModify;
+import com.palantir.gradle.suppressibleerrorprone.modes.common.ModifyCheckApiOption.MustModify;
+import com.palantir.gradle.suppressibleerrorprone.modes.common.ModifyCheckApiOption.NoEffect;
+import java.util.Collection;
+import java.util.Set;
+import java.util.function.Predicate;
+import one.util.streamex.StreamEx;
+
+/**
+ * suppressible-error-prone is powered by an artifact transform that modifies the error_prone_check_api jar.
+ * Some modes must not run with this jar modified, some must run with it modified (and then some may or may not need
+ * the VisitorState class modified), and some don't care. This class represents these states and allows combining them
+ * sensibly.
+ */
+public sealed interface ModifyCheckApiOption permits DoNotModify, NoEffect, MustModify {
+    /**
+     * The error_prone_check_api jar must not be modified.
+     */
+    static ModifyCheckApiOption doNotModify() {
+        return DoNotModify.INSTANCE;
+    }
+
+    /**
+     * It doesn't matter whether the error_prone_check_api jar is modified or not. This will have no effect on the
+     * decision made.
+     */
+    static ModifyCheckApiOption noEffect() {
+        return NoEffect.INSTANCE;
+    }
+
+    /**
+     * The error_prone_check_api jar must be modified to allow `for-rollout:` suppressions to work.
+     */
+    static ModifyCheckApiOption mustModify() {
+        return new MustModify(false);
+    }
+
+    /**
+     * The error_prone_check_api jar must be modified to allow `for-rollout:` suppressions to work, and VisitorState
```

Comment:
```suggestion
     * The{@code error_prone_check_api} jar must be modified to allow {@code for-rollout:} suppressions to work, and {@code VisitorState}
```

I'm not sure if `VisitorState` is on the classpath, if so `{@link VisitorState}` instead?